config/management/operational/src/validator_state.rs

```diff
@@ -2,67 +2,123 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::{json_rpc::JsonRpcClientWrapper, validator_config::DecryptedValidatorConfig};
-use diem_global_constants::{CONSENSUS_KEY, OWNER_ACCOUNT};
-use diem_management::error::Error;
+use diem_global_constants::{
+    CONSENSUS_KEY, FULLNODE_NETWORK_KEY, OWNER_ACCOUNT, VALIDATOR_NETWORK_KEY,
+};
+use diem_management::{
+    config::ConfigPath, error::Error, secure_backend::ValidatorBackend, storage::to_x25519,
+};
 use serde::Serialize;
 use structopt::StructOpt;
 
-#[derive(Debug, Default, Serialize)]
+#[derive(Debug, Serialize)]
 pub struct VerifyValidatorStateResult {
-    /// Check if a validator is in the latest validator set on-chain.
-    pub in_validator_set: Option<bool>,
     /// Check if the consensus key held in secure storage matches
     /// that registered on-chain for the validator.
     pub consensus_key_match: Option<bool>,
+
+    /// If the consensus_key_match is true, check if the key is unique
+    pub consensus_key_unique: Option<bool>,
```

Comment:
Why do we need the `consensus_key_match` to be true to check this? It seems orthogonal, no? In my mind, the more checks we can do the better, i.e., checks should only be avoided if they aren't possible/don't make sense any more.

Reply:
Updated to check `consensus_key_unique` even if consensus key does not match.

You're right that this `consensus_key_unique` check should be orthogonal to `consensus_key_match`. As you mentioned, It is useful when a new node (who is not in the validator set yet) wants to check if its key is unique.
(At first, I only thought about the case where a node wants to check if it is already in the set, and then if its key is unique among the nodes in the set.)